bifromq-inbox/bifromq-inbox-server/src/test/java/com/baidu/bifromq/inbox/server/InboxWriterPipelineTest.java

```diff
@@ -0,0 +1,211 @@
+
+        SendReply sendReply = writerPipeline.handleRequest("_", sendRequest(matchInfo)).join();
+        assertEquals(sendReply, mockSendReply);
+    }
+
+    private MatchInfo createMatchInfo() {
+        return MatchInfo.newBuilder()
+                .setTopicFilter("/foo/+")
+                .setIncarnation(1L)
+                .setReceiverId(receiverId("foo", 1L))
+                .build();
+    }
+
+    private SendReply createSendReply(DeliveryResult.Code code, MatchInfo matchInfo) {
+        return SendReply.newBuilder()
+                .setReqId(1)
+                .setReply(DeliveryReply.newBuilder()
+                        .putResult(TENANT_ID, DeliveryResults.newBuilder()
+                                .addResult(DeliveryResult.newBuilder()
+                                        .setMatchInfo(matchInfo)
+                                        .setCode(code).build()).build())
+                        .build())
+                .build();
+    }
+
+
+    @Test
+    public void testConstructorDirectMemoryUsageCatch() {
+        // Equal, used to trigger subsequent comparisons
+        when(memUsage.nettyDirectMemoryUsage()).thenReturn(IngressSlowDownDirectMemoryUsage.INSTANCE.get() + 0.1f);
+        when(memUsage.heapMemoryUsage()).thenReturn(IngressSlowDownHeapMemoryUsage.INSTANCE.get() - 0.1f);
+        testMemoryUsageThresholdExceed();
+    }
+
+    @Test
+    public void testConstructorHeapMemoryUsageCatch() {
+        // Equal, used to trigger subsequent comparisons
+        when(memUsage.nettyDirectMemoryUsage()).thenReturn(IngressSlowDownDirectMemoryUsage.INSTANCE.get() - 0.1f);
+        when(memUsage.heapMemoryUsage()).thenReturn(IngressSlowDownHeapMemoryUsage.INSTANCE.get() + 0.1f);
+        testMemoryUsageThresholdExceed();
+    }
+
+    @Test
+    public void testConstructorHeapMemoryUsageAllNotCatch() {
+        // Equal, used to trigger subsequent comparisons
+        when(memUsage.nettyDirectMemoryUsage()).thenReturn(IngressSlowDownDirectMemoryUsage.INSTANCE.get() - 0.1f);
+        when(memUsage.heapMemoryUsage()).thenReturn(IngressSlowDownHeapMemoryUsage.INSTANCE.get() - 0.1f);
+        testMemoryUsageThresholdExceed();
+    }
+
+
+    public void testMemoryUsageThresholdExceed() {
+
+
+        when(inboxWriter.handle(any())).thenReturn(CompletableFuture.completedFuture(SendReply.getDefaultInstance()));
+        doNothing().when(fetcherSignaler).afterWrite(any(), any());
+        setLocalMemUsage(memUsage);
+
+        SendRequest sendRequest = SendRequest.getDefaultInstance();
+
+        InboxWriterPipeline writerPipeline = new InboxWriterPipeline(fetcherSignaler, inboxWriter, responseObserver);
+        writerPipeline.onNext(sendRequest);
+        SendReply sendReply = writerPipeline.handleRequest("_", sendRequest).join();
+        assertEquals(sendReply, SendReply.getDefaultInstance());
+    }
+
+
+    public static void setLocalMemUsage(MemUsage memUsage) {
```

Comment:
don't use reflection to mock static method, use mockito to mock static method instead.